pkg/pillar/zedUpload/azureutil/azure.go

```diff
@@ -242,3 +244,91 @@ func GetAzureBlobMetaData(accountName, accountKey, containerName, remoteFile str
 	stringHex := hex.EncodeToString(decodedString)
 	return blob.Properties.ContentLength, stringHex, nil
 }
+
+// GenerateBlobSasURI is used to generate the URI which can be used to access the blob until the the URI expries
+func GenerateBlobSasURI(accountName, accountKey, containerName, remoteFile string, httpClient *http.Client) (string, error) {
+	c, err := NewClient(accountName, accountKey, httpClient)
+	if err != nil {
+		return "", err
+	}
+	blobClient := c.GetBlobService()
+	container := blobClient.GetContainerReference(containerName)
+	containerExists, _ := container.Exists()
+	if !containerExists {
+		return "", err
+	}
+	blob := container.GetBlobReference(remoteFile)
+	exists, err := blob.Exists()
+	if err != nil {
+		return "", err
+	}
+	if !exists {
+		return "", fmt.Errorf("blob doesn't exist")
+	}
+	options := storage.BlobSASOptions{}
+	options.Read = true
+	options.Start = time.Now()
+	options.Expiry = time.Now().Add(time.Hour * 2)
```

Comment:
Why validity is needed for 2 hours?

Reply:
just to align with the session time out, it is 2 hours, it makes more sense if we pass that as an argument instead of hard coding. will change accordingly.

Reply:
My suggestion is to keep the validity as short as possible.

Reply:
Initial Implementation was with 10mins, but Sahil is insisting to keep inline with session time, again we discuss with concerned people and we can conclude upon this.